### PR TITLE
Update taxonomy props

### PR DESCRIPTION
**Pull request by ramirobg94**

### CODE REVIEW

File: webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    setState(State.assocTaxonomy(State.assocTaxonomyProp({ key, value })(taxonomy))(state))
+
+    const action = async () => {
+      dispatch({ type: TaxonomiesActions.taxonomyPropUpdate, taxonomy, key, value })
+      const { data } = await axios.put(`/api/survey/${surveyId}/taxonomies/${Taxonomy.getUuid(taxonomy)}`, {
+        key,
+        value,
+      })
+      dispatch({ type: TaxonomiesActions.taxonomiesUpdate, taxonomies: data.taxonomies })
```
Comment (ramirobg94) [inline webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js:27]: maybe we can discuss if we can do that when the component is unmounted

File: webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { State } from '../state'
+
+export const useUpdate = ({ setState }) => {
+  const dispatch = useDispatch()
+  const surveyId = useSurveyId()
+
+  return useCallback(({ key, value, state }) => {
+    const taxonomy = State.getTaxonomy(state)
+
+    setState(State.assocTaxonomy(State.assocTaxonomyProp({ key, value })(taxonomy))(state))
```
Comment (minotogna) [inline webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js:19]: maybe some variable would make the code more readable

Comment (minotogna) [reply webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js:19]: or a pipe

Comment (ramirobg94) [reply webapp/views/App/views/Designer/TaxonomyDetail/store/actions/useUpdate.js:19]: sure I also felt the same :)

**Pull request merged**